### PR TITLE
PCA model

### PR DESCRIPTION
**Pull request by aniketsharma00411**

### CODE REVIEW

File: include/all.hpp [hunk 1]
```diff
@@ -1,9 +1,11 @@
 #ifndef _all_hpp_
 #define _all_hpp_
 
+#include <Eigen/SVD>
 #include <Matrix.hpp>
 #include <assert.h>
 #include <iostream>
+#include <math.h>
```
Comment (ashok-arora) [inline include/all.hpp]: `math.h` is a C header library, it should be replaced by it's C++ counterpart `cmath` and likewise for `assert.h` and `stdlib.h`.

File: include/PCA.hpp [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+class PCA {
+  private:
+    bool is_fit = false;
+    int n_components;
+
+    Eigen::MatrixXd convert2eigen(Matrix);
+    Matrix convert2matrix(Eigen::MatrixXd);
+
+  public:
+    Matrix Ureduce, Sigma;
```
Comment (ashok-arora) [inline include/PCA.hpp]: Can this be moved to private?

File: examples/PCA.cpp [hunk 3]
```diff
@@ -0,0 +1,48 @@
+// Include necessary header files
+#include <Matrix.hpp>
+#include <PCA.hpp>
+#include <matplotlibcpp.hpp>
```
Comment (ashok-arora) [inline examples/PCA.cpp]: Is there any plot that can be added for visualization?

File: include/PCA.hpp (hunk 2 above)
Comment (aniketsharma00411) [reply include/PCA.hpp:15]: Na

File: examples/PCA.cpp (hunk 3 above)
Comment (aniketsharma00411) [reply examples/PCA.cpp]: Na

**Review by ashok-arora — APPROVED** [review]

**Pull request merged**